app/controllers/shopify_app/callback_controller.rb

```diff
@@ -69,7 +69,7 @@ def start_user_token_flow?
       if jwt_request?
         false
       else
-        ShopifyApp::SessionRepository.user_storage.present? && user_session.blank?
+        ShopifyApp::SessionRepository.user_storage.present? && associated_user.blank?
```

Comment:
This changes the meaning of this code, right? Before it would decide to start the user flow if the app had user storage configured and a user session didn't exist. Now it's going to start a user token flow when user storage is present and it hasn't just completed a user token flow, because that's the only time when associated_user will be populated.

Does that matter?

Reply:
I'm curious about whether this can ever be the case:

* User storage is configured
* A `user_session` is not blank
* `associated_user` is blank

Reply:
That is the case when you're arriving at the callback controller after fetching an offline/shop token. But currently that case isn't likely to exist.